src/utils.rs

```diff
@@ -114,6 +114,45 @@ impl Color {
     }
 }
 
+/// A Color used in a gradient
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct ColorGradient {
+    red: u8,
+    green: u8,
+    blue: u8,
+}
+
+impl ColorGradient {
+    fn new(red: u8, green: u8, blue: u8) -> Self {
+        ColorGradient {
+            red: red.min(255),
+            green: green.min(255),
+            blue: blue.min(255),
+        }
+    }
+
+    fn from_hex(hex: &str) -> Self {
+        // Remove '#' if present
+        let s = hex.trim_start_matches('#');
+
+        // 6 char hex
+        let channels: Vec<u8> = (0..5)
```

Comment:
Can we collect into a `[u8; 3]` instead of into a `Vec` directly?

Reply:
Did a little something to use a fixed size array